subsys/bluetooth/audio/has_client.c

```diff
@@ -82,11 +88,195 @@ static enum bt_has_capabilities get_capabilities(const struct has_inst *inst)
 	return caps;
 }
 
+union preset_record {
+	struct bt_has_preset_record record;
+	struct {
+		enum bt_has_properties properties;
+		char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	};
+};
+
+static void handle_read_preset_rsp(struct bt_conn *conn, struct net_buf_simple *buf)
+{
+	struct has_inst *inst = inst_by_conn(conn);
+	const struct bt_has_cp_read_preset_rsp *rsp;
+	union preset_record record;
+	size_t name_len;
+
+	BT_DBG("conn %p buf %p", (void *)conn, buf);
+
+	if (buf->len < sizeof(*rsp)) {
+		BT_ERR("malformed PDU");
+		return;
```

Comment:
Given that `preset_event` has an error field, should we call the callback here?

Reply:
Applies multiple places

Reply:
I don't feel this information could be useful